velox/connectors/hive/HiveConfig.cpp

```diff
@@ -62,166 +52,161 @@ std::string HiveConfig::insertExistingPartitionsBehaviorString(
 }
 
-// static
-std::optional<std::string> HiveConfig::s3SecretKey(const Config* config) {
-  if (config->isValueExists(kS3AwsSecretKey)) {
-    return config->get(kS3AwsSecretKey).value();
+std::optional<std::string> HiveConfig::s3AccessKey() const {
+  if (config_->isValueExists(kS3AwsAccessKey)) {
+    return config_->get(kS3AwsAccessKey).value();
   }
   return {};
 }
 
-// static
-std::optional<std::string> HiveConfig::s3IAMRole(const Config* config) {
-  if (config->isValueExists(kS3IamRole)) {
-    return config->get(kS3IamRole).value();
+std::optional<std::string> HiveConfig::s3SecretKey() const {
+  if (config_->isValueExists(kS3AwsSecretKey)) {
+    return config_->get(kS3AwsSecretKey).value();
   }
   return {};
 }
 
-// static
-std::string HiveConfig::s3IAMRoleSessionName(const Config* config) {
-  return config->get(kS3IamRoleSessionName, std::string("velox-session"));
-}
-
-// static
-std::string HiveConfig::gcsEndpoint(const Config* config) {
-  return config->get<std::string>(kGCSEndpoint, std::string(""));
+std::optional<std::string> HiveConfig::s3IAMRole() const {
+  if (config_->isValueExists(kS3IamRole)) {
+    return config_->get(kS3IamRole).value();
+  }
+  return {};
 }
 
-// static
-std::string HiveConfig::gcsScheme(const Config* config) {
-  return config->get<std::string>(kGCSScheme, std::string("https"));
+std::string HiveConfig::s3IAMRoleSessionName() const {
+  return config_->get(kS3IamRoleSessionName, std::string("velox-session"));
 }
 
-// static
-std::string HiveConfig::gcsCredentials(const Config* config) {
-  return config->get<std::string>(kGCSCredentials, std::string(""));
+std::string HiveConfig::gcsEndpoint() const {
+  return config_->get<std::string>(kGCSEndpoint, std::string(""));
 }
 
-// static.
-bool HiveConfig::isOrcUseColumnNames(const Config* config) {
-  return config->get<bool>(kOrcUseColumnNames, false);
+std::string HiveConfig::gcsScheme() const {
+  return config_->get<std::string>(kGCSScheme, std::string("https"));
 }
 
-// static.
-bool HiveConfig::isFileColumnNamesReadAsLowerCase(const Config* config) {
-  return config->get<bool>(kFileColumnNamesReadAsLowerCase, false);
+std::string HiveConfig::gcsCredentials() const {
+  return config_->get<std::string>(kGCSCredentials, std::string(""));
 }
 
-// static.
-int64_t HiveConfig::maxCoalescedBytes(const Config* config) {
-  return config->get<int64_t>(kMaxCoalescedBytes, 128 << 20);
+bool HiveConfig::isOrcUseColumnNames(const Config* session) const {
+  if (session->isValueExists(kOrcUseColumnNamesSession)) {
+    return session->get<bool>(kOrcUseColumnNamesSession).value();
+  }
+  return config_->get<bool>(kOrcUseColumnNames, false);
 }
 
-// static.
-int32_t HiveConfig::maxCoalescedDistanceBytes(const Config* config) {
-  return config->get<int32_t>(kMaxCoalescedDistanceBytes, 512 << 10);
+bool HiveConfig::isFileColumnNamesReadAsLowerCase(const Config* session) const {
+  if (session->isValueExists(kFileColumnNamesReadAsLowerCaseSession)) {
```

Comment:
@kewang1024 @majetideepak  @xiaoxmeng @mbasmanova 

Apart from the issue mentioned by @zhztheplayer [here](https://github.com/facebookincubator/velox/pull/7725/files#r1424760919). There is one another issue here. We have currently set `kFileColumnNamesReadAsLowerCase `to true in Gluten, but it didn't take effect. We must set `kFileColumnNamesReadAsLowerCaseSession `to true in order for it to work. Based on the code analysis, it seems that if the `kFileColumnNamesReadAsLowerCaseSession `parameter is not set, it will read the `kFileColumnNamesReadAsLowerCase `configuration. However, it appears that it is not taking effect.

Reply:
@JkSelf Would you create GitHub issue to explain the problem you are facing? We can discuss it there.

Reply:
`kFileColumnNamesReadAsLowerCase` should be set when a HiveConnector is created, `kFileColumnNamesReadAsLowerCaseSession` should be set per query basis in the session property. 

can you show me where your code is not working?

Reply:
@kewang1024 The related code is [here](https://github.com/oap-project/gluten/blob/main/cpp/velox/compute/WholeStageResultIterator.cc#L427). We set `kFileColumnNamesReadAsLowerCase` config to true without this PR. And after this PR, we need to set `kFileColumnNamesReadAsLowerCaseSession` to true and then the unit test can pass in Gluten. It seems the `kFileColumnNamesReadAsLowerCase` is not longer taking effect in our case.